src/mintpy/prep_nisar.py

```diff
@@ -263,20 +274,47 @@ def read_and_interpolate_geometry(gunw_file, dem_file, xybbox):
     # dem_subset_array = np.zeros((subset_rows, subset_cols), dtype=raster_array.dtype)
     Y_2d, X_2d = np.meshgrid(ycoord, xcoord, indexing='ij')
 
-    if not src_epsg == dst_epsg:
-        coord_transform = Transformer.from_crs(dst_epsg, src_epsg, always_xy=True)
+    if not dem_src_epsg == dst_epsg:
+        coord_transform = Transformer.from_crs(dst_epsg, dem_src_epsg, always_xy=True)
```

Comment:
This comment did not make it to my review so posting it here again. It looks like you want to check if the epsg of the input DEM is different from GUNW and if they are different then project the DEM to the GUNW epsg and to its exact grid or a given bounding box. I'm wondering why you are not letting gdal.warp do that for you. Something like this should work:
```
gdal.Warp(output_dem_path, 
          src_ds, 
          format='GTiff', 
          outputBounds=bbox, 
          xRes=x_res, yRes=y_res,
          srcSRS=source_epsg, 
          dstSRS=target_epsg,
          resampleAlg=gdal.GRA_Bilinear,
          options=['COMPRESS=DEFLATE']) 
```

Reply:
Yes thank you, that is a more proper way. I will change it